crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -158,7 +159,15 @@ impl BaseRoomInfo {
     pub fn handle_stripped_state_event(&mut self, ev: &AnyStrippedStateEvent) -> bool {
         match ev {
             AnyStrippedStateEvent::RoomEncryption(encryption) => {
-                self.encryption = Some(encryption.content.clone());
+                if let Some(algorithm) = &encryption.content.algorithm {
+                    let content = assign!(RoomEncryptionEventContent::new(algorithm.clone()), {
+                        rotation_period_ms: encryption.content.rotation_period_ms,
+                        rotation_period_msgs: encryption.content.rotation_period_msgs,
+                    });
+                    self.encryption = Some(content);
+                } else {
+                    // FIXME: encryption event exists but is redacted
```

Comment:
In that case, let's set the default rotation periods.

Reply:
That's not the problem, the problem is unknown algorithm.